src/include/api.h

```diff
@@ -97,8 +97,13 @@
         bool __set_err = true;                                                              \
         const char *(cfg)[] = {WT_CONFIG_BASE(s, struct_name##_##func_name), config, NULL}; \
         API_SESSION_INIT(s, struct_name, func_name, dh);                                    \
-        if ((config) != NULL)                                                               \
-    WT_ERR(__wt_config_check((s), WT_CONFIG_REF(s, struct_name##_##func_name), (config), 0))
+        if (cfg[1] != NULL) {                                                               \
```

Comment:
This deserves a comment I think. Maybe something like:
/*
 * Optimize configuration checking. Check for the user sending in an empty configuration string
 * that is known to be in array element 1. If it is empty then set the overall element to NULL to save
 * additional checking.
 * /

Reply:
I added a comment, slightly different than your wording.